spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -269,174 +285,341 @@ def get_num_frames(self):
     def get_channel_ids(self):
         return self.channel_ids.tolist()
 
-    @staticmethod
-    def add_devices(recording, nwbfile, metadata):
-        # Devices
-        if 'Ecephys' not in metadata:
-            metadata['Ecephys'] = dict()
-        if 'Device' not in metadata['Ecephys']:
-            metadata['Ecephys']['Device'] = [{'name': 'Device'}]
-        # Tests if devices exist in nwbfile, if not create them from metadata
-        for dev in metadata['Ecephys']['Device']:
-            if dev['name'] not in nwbfile.devices:
-                nwbfile.create_device(name=dev['name'])
+    # The strategy for each add_* method is to form a dictionary of keyword
+    # arguments that will be passed into nwbfile.create_*, with user-specified
+    # metadata always overriding all internal recording information. If no
+    # metadata is passed, internal channel properties will be used when
+    # available, otherwise set all to defaults.
 
+    @staticmethod
+    def add_devices(recording: se.RecordingExtractor, nwbfile: NWBFile,
```

Comment:
@CodyCBakerPhD can you set the `nwbfile=None` and remove `: NWBFile`? This would require `pynwb` to be a requirement of `spikeextractors`. You can add a check in the function for this.

Same for the other `add` functions